extensions/MozChangeField/lib/CanConfirm.pm

```diff
@@ -0,0 +1,55 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::MozChangeField::CanConfirm;
+
+use 5.10.1;
+use Moo;
+
+use Bugzilla::Constants;
+
+sub evaluate_change {
+  my ($self, $args) = @_;
+
+  my $bug          = $args->{'bug'};
+  my $field        = $args->{'field'};
+  my $new_value    = $args->{'new_value'};
+  my $old_value    = $args->{'old_value'};
+  my $priv_results = $args->{'priv_results'};
+  my $canconfirm   = $args->{'canconfirm'};
+  my $editbugs     = $args->{'editbugs'};
+
+  # Canconfirm is really "cantriage"; users with canconfirm can also mark
+  # bugs as DUPLICATE, WORKSFORME, and INCOMPLETE.
+  if (
```

Comment:
The `$field eq 'bug_status' && is_open_state($old_value) && !is_open_state($new_value)` check from the original code is missing.